front/lib/plans/usage/index.ts

```diff
@@ -0,0 +1,74 @@
+import type { LightWorkspaceType, Result } from "@dust-tt/types";
+import { assertNever, Err, Ok } from "@dust-tt/types";
+import type Stripe from "stripe";
+
+import { reportMonthlyActiveUsers } from "@app/lib/plans/usage/mau";
+import type {
+  InvalidRecurringPriceError,
+  SupportedReportUsage,
+} from "@app/lib/plans/usage/types";
+import {
+  InvalidReportUsageError,
+  isSupportedReportUsage,
+  REPORT_USAGE_METADATA_KEY,
+} from "@app/lib/plans/usage/types";
+
+function getUsageToReportForSubscriptionItem(
+  item: Stripe.SubscriptionItem
+): Result<SupportedReportUsage | null, InvalidReportUsageError> {
+  const usageToReport = item.price.metadata[REPORT_USAGE_METADATA_KEY];
+
+  if (!usageToReport) {
+    return new Ok(null);
+  }
+
+  if (isSupportedReportUsage(usageToReport)) {
+    return new Ok(usageToReport);
+  }
+
+  return new Err(new InvalidReportUsageError());
+}
+
+export async function reportUsageForSubscriptionItems(
+  stripeSubscription: Stripe.Subscription,
+  workspace: LightWorkspaceType
+): Promise<
+  Result<undefined, InvalidRecurringPriceError | InvalidReportUsageError>
+> {
+  const { data: subscriptionItems } = stripeSubscription.items;
+
+  for (const item of subscriptionItems) {
+    const usageToReportRes = getUsageToReportForSubscriptionItem(item);
+    if (usageToReportRes.isErr()) {
+      return new Err(usageToReportRes.error);
+    }
+
+    const usageToReport = usageToReportRes.value;
+    if (!usageToReport) {
+      return new Ok(undefined);
+    }
+
+    switch (usageToReport) {
```

Comment:
checking: is the switch here to prepare for having a "PER_SEAT" usage report in soon?

Reply:
That's correct. Coming very very soon.